src/channel.rs

```diff
@@ -43,15 +43,23 @@ impl Channel {
     }
     pub fn close(&mut self, reply_code: u16, reply_text: String) {
         let close = &channel::Close {reply_code: reply_code, reply_text: reply_text, class_id: 0, method_id: 0};
-        let _: channel::CloseOk = self.rpc(close, "channel.close-ok").ok().expect("Error closing connection");
+        let call: Option<channel::CloseOk> = self.rpc(close, "channel.close-ok").ok();
+        if call.is_none() {
+            error!("Error closing connection")
+        }
     }
 
     pub fn read(&self) -> AMQPResult<Frame> {
-        self.receiver.recv().ok().expect("Error reading packet from channel")
+        self.receiver.recv().map_err(|_|
+            AMQPError::Protocol("Error reading packet from channel".to_owned())
+        ).and_then(|frame| frame)
     }
 
     pub fn write(&mut self, frame: Frame) {
-        self.connection.write(frame).ok().expect("Error writing packet to connection");
+        match self.connection.write(frame) {
```

Comment:
Are you sure we should't do anything when we fail writing frame?
I think it's something we should definitely handle or at least blow up (as it's currently happening), otherwise the behaviour will be very weird, as the code will not throw any errors, but nothing works.